File: PP.Lab3J/Producer.java
public class Producer implements Runnable {
    private final WorkInfo info;
    private final BoundedStorage storage;

    public Producer(WorkInfo info, BoundedStorage storage) {
        this.info = info;
        this.storage = storage;
    }

    @Override
    public void run() {
        for (int i = 0; i < info.getQuota(); i++) {
            try { Thread.sleep(500); } catch (InterruptedException e) { Thread.currentThread().interrupt(); }
            storage.put(i, info.getId());
        }
    }
}

File: PP3.LabK/src/Producer.kt
import java.util.concurrent.CountDownLatch

class Producer(
    private val info: WorkerInfo,
    private val storage: BoundedStorage,
    private val latch: CountDownLatch
) : Runnable {

    override fun run() {
        try {
            for (i in 1..info.count) {
                val item = "Item $i from Producer #${info.id}"
                storage.put(item, info.id)
                Thread.sleep(500)
            }
            println(">>> Producer ${info.id} has finished their quota.")
        } catch (e: InterruptedException) {
            Thread.currentThread().interrupt()
        } finally {
            latch.countDown()
        }
    }
}